lib/ansible/galaxy/collection/concrete_artifact_manager.py

```diff
@@ -656,14 +656,8 @@ def _get_json_from_installed_dir(
     try:
         with open(b_json_filepath, 'rb') as manifest_fd:
             b_json_text = manifest_fd.read()
-    except (IOError, OSError):
-        raise LookupError(
-            "The collection {manifest!s} path '{path!s}' does not exist.".
-            format(
-                manifest=filename,
-                path=to_native(b_json_filepath),
-            )
-        )
+    except OSError as ex:
+        raise LookupError(f"The collection {filename!r} path {to_native(b_json_filepath)!r} does not exist.") from ex
```

Comment:
We can't eliminate the conversion here yet, but we can at least use `to_text`:

```suggestion
        raise LookupError(f"The collection {filename!r} path {to_text(b_json_filepath)!r} does not exist.") from ex
```

There's no reason for new or updated code to use `to_native`, since it's just an alias for `to_text`.